server/src/main/java/org/elasticsearch/index/engine/SearchOnlyEngine.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.engine;
+
+import org.apache.lucene.index.SegmentInfos;
+import org.apache.lucene.search.SearcherManager;
+import org.apache.lucene.store.AlreadyClosedException;
+import org.elasticsearch.common.lease.Releasable;
+import org.elasticsearch.common.lease.Releasables;
+import org.elasticsearch.common.util.concurrent.ReleasableLock;
+import org.elasticsearch.core.internal.io.IOUtils;
+import org.elasticsearch.index.seqno.SeqNoStats;
+import org.elasticsearch.index.translog.Translog;
+import org.elasticsearch.index.translog.TranslogStats;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+
+/**
+ * A minimal engine that does not accept writes, and always points stats, searcher to the last commit.
+ */
+public final class SearchOnlyEngine extends Engine {
+    private final SegmentInfos lastCommittedSegmentInfos;
+    private final SeqNoStats seqNoStats;
+    private final TranslogStats translogStats;
+    private final SearcherManager searcherManager;
+    private final Searcher lastCommitSearcher;
+
+    public SearchOnlyEngine(Engine engine) throws IOException {
+        super(engine.config());
+        engine.refresh("lockdown");
+        this.seqNoStats = engine.getSeqNoStats(engine.getLastSyncedGlobalCheckpoint());
+        this.translogStats = engine.getTranslogStats();
+        this.lastCommittedSegmentInfos = engine.getLastCommittedSegmentInfos();
+        Searcher searcher = engine.acquireSearcher("lockdown", SearcherScope.INTERNAL);
```

Comment:
can you leave a comment here that we keep a reference to the store implicitly through the searcher? I do wonder if we should make it explicit